tests/test_views.py

```diff
@@ -2848,25 +2848,6 @@ def _make_pointer_only_user_can_see(self, user, project, save=False):
         node = ProjectFactory(creator=user)
         project.add_pointer(node, auth=Auth(user=user), save=save)
 
-    def test_pointer_list_write_contributor_can_remove_private_component_entry(self):
```

Comment:
I'm not sure where to go with this. This PR has removed most, possibly all, visibility to "Private" links/components. My understanding is that a user without at least read permissions should not know about "Private" links/components, even their existence.
  This policy may be somewhat academic, but it has a practical advantage for me in this PR as I can skip "Private" breaks in the tree and show previously hidden sub-trees. I've updated the components widget to match the tree view.
  As this PR stands, I won't be able to make this test pass. Maybe there needs to be a discussion about how we want to handle this edge case. It seems like it might be a security/user experience question. Who can delete a component/link when no one has permissions to it, and where in the interface it should happen.